htdocs/install/mysql/migration/20.0.0-21.0.0.sql

```diff
@@ -87,6 +87,7 @@ ALTER TABLE llx_c_hrm_public_holiday ADD UNIQUE INDEX uk_c_hrm_public_holiday(en
 ALTER TABLE llx_c_hrm_public_holiday ADD UNIQUE INDEX uk_c_hrm_public_holiday2(entity, fk_country, dayrule, day, month, year);
 
 ALTER TABLE llx_societe_account ADD COLUMN date_last_reset_password datetime after date_previous_login;
+ALTER TABLE llx_user_rib ADD COLUMN default_rib smallint NOT NULL DEFAULT 1;
```

Comment:
If you set DEFAULT 1 in sql, it means when we will be able to insert new lines, they will have default set automatically.
May be you can set "DEFAULT NULL".
Then run UPDATE set default_rib = 1 where default_rib IS NULL 
so all existing lines, at migration time, will have default_rib to 1, and then later, when we will enter a new bank, it will not be set as the default.

Reply:
Thank you for your time!

After discussing with the team and our client we realized that issue https://github.com/Dolibarr/dolibarr/issues/30810 isn't about **user** ribs, so doing this is bonus.
With that in mind and since I'm running out of time this sprint, I won't be able to properly develop a work logic around **user** ribs soon. (ie: setting which ribs are to be used by default for each user, allowing to change it, etc.)

We then decided to do the following:
- I reverted the code NOT to use "default =1" in the sql request
- I STILL create the column and set the default to 0 (see below for explanation).
- I updated the TODO comment to say that the column has been created if someone wants to do the work logic.
I may end up doing it in the future, but I don't want to promise it so at least it's one step done


**Why not setting default to NULL ?**
--> because it's supposed to be working the same way as default_rib from societe_rib, which is NOT NULL, and by default 0

**Why not using UPDATE to set 1 to all existing so we can still add the sql "AND sr.default_rib = 1"?**
--> It would work, but only at first. Since we have no work logic for users yet, creating a new user would set its value to 0, while having every old user to 1, with no way to modify it. In the sql request it wouldn't be able to select the new user.